### PR TITLE
Audio buffer starvation detection

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by huffman** [comment]
Fixing merge conflicts

**Comment by huffman** [comment]
Fixed

**Comment by huffman** [comment]
Ok, fixed the `DependencyManager` usage

File: interface/src/Audio.cpp [hunk 1]
```diff
@@ -949,8 +955,24 @@ void Audio::outputNotify() {
     if (recentUnfulfilled > 0) {
         qDebug() << "WARNING --- WE HAD at least:" << recentUnfulfilled << "recently unfulfilled readData() calls";
```
Comment (ZappoMan) [inline interface/src/Audio.cpp]: let's get rid of this warning maybe? or at least add it to the suppress repeat list

File: interface/src/Audio.cpp [hunk 2]
```diff
@@ -949,8 +955,24 @@ void Audio::outputNotify() {
...
 
-        // TODO: Ryan Huffman -- add code here to increase the AUDIO_OUTPUT_BUFFER_SIZE_FRAMES... this code only
-        // runs in cases where the audio device requested data samples, and ran dry because we couldn't fulfill the request
+        if (_outputStarveDetectionEnabled) {
+            quint64 now = QDateTime::currentMSecsSinceEpoch();
```
Comment (ZappoMan) [inline interface/src/Audio.cpp]: use usecTimestampNow()

File: interface/src/Audio.cpp (hunk 1 above)
Comment (huffman) [inline interface/src/Audio.cpp]: What is the suppress repeat list?

Comment (huffman) [inline interface/src/Audio.cpp]: How do I add it to the list, rather.

File: interface/src/Audio.cpp [hunk 3]
```diff
@@ -947,10 +953,24 @@ bool Audio::switchInputToAudioDevice(const QAudioDeviceInfo& inputDeviceInfo) {
 void Audio::outputNotify() {
     int recentUnfulfilled = _audioOutputIODevice.getRecentUnfulfilledReads();
     if (recentUnfulfilled > 0) {
-        qDebug() << "WARNING --- WE HAD at least:" << recentUnfulfilled << "recently unfulfilled readData() calls";
```
Comment (ZappoMan) [inline interface/src/Audio.cpp]: if you want to keep this, but want to make "repeated versions" suppressed do something like this...

```
        static QString repeatedMessage = LogHandler::getInstance().addRepeatedMessageRegex("WARNING --- WE HAD at least: ?\\d+ recently unfulfilled readData() calls");

```

Before you call the qDebug() << "..."

Comment (huffman) [inline interface/src/Audio.cpp]: Cool - I ended up removing it, but that's useful for the future

**Pull request merged**